usr/share/rear/layout/save/GNU/Linux/200_partition_layout.sh

```diff
@@ -30,14 +30,21 @@ extract_partitions() {
 
     declare path sysfs_path
     if [[ ${#sysfs_paths[@]} -eq 0 ]] ; then
-        ### try to find partitions like /dev/mapper/datalun1p1
         if [[ ${device/mapper//} != ${device} ]] ; then
```

Comment:
@schabrolles 
some time ago I was already staring at that
<pre>
    if [[ ${device/mapper//} != ${device} ]] ; then
</pre>
condition and now I am again staring at it
but it looks so strange to my "hawk eyes".
Could you perhaps provide a comment what is actually tested here
or perhaps even better if possible change it into understandable code?
I guess it tests if $device contains the `mapper` sub-string anywhere
but I assume the intent is to test if $device starts with `dev/mapper`
or at least if $device contains the `/mapper/` sub-string.
The current oversophisticatedly witten code just fails e.g. for
<pre>
# device="/dev/mymapperstuff/mymapperpath"

# echo ${device/mapper//}
/dev/my/stuff/mymapperpath

# if [[ ${device/mapper//} != ${device} ]] ; then echo Yes ; fi
Yes
</pre>
while plain and simple dumb direct testing
for the actually intended sub-sting "just works":
<pre>
# device="/dev/mymapperstuff/mymapperpath"

# if [[ $device = *'/mapper/'* ]] ; then echo Yes ; else echo No ; fi
No

# device="/dev/mapper/mymapperpath"

# if [[ $device = *'/mapper/'* ]] ; then echo Yes ; else echo No ; fi
Yes
</pre>
(I tested that this works even on SLES10)

Reply:
I'm not the author of this line. According to git, @jhoekx wrote it 6 years ago... 
I usually try to not modify what is currently working (avoiding side effect), but I agree with you, I think we can make it easier to read as the purpose is to detect if the device is managed by device-mapper (in `/dev/mapper`)